manager/container.go

```diff
@@ -397,27 +399,44 @@ func (c *containerData) updateLoad(newLoad uint64) {
 	}
 }
 
-func (c *containerData) updateStats() error {
+func (c *containerData) updateStats() (nextCollectionTime time.Time, err error) {
+	nextCollectionTime, metrics, metricsErr := c.collectorManager.Collect()
+	if metricsErr != nil {
+		// Ignore errors if the container is dead.
+		if !c.handler.Exists() {
+			return nextCollectionTime, nil
+		}
+
+		// Metrics may be partially populated, push those before we return an error.
+		metricsErr = fmt.Errorf("%v, continuing to push metrics", metricsErr)
+	}
+	if metrics == nil {
```

Comment:
This would stop getting all regular stats if we don't have any custom stats. Any reason we're integration collection here instead of the callsite above? (the one we commented out)